### PR TITLE
Timeline

### PR DESCRIPTION
**Pull request by aslakhol**
The timeline component from the online.ntnu.no/splash

### CODE REVIEW

File: src/components/timeline/Day.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+};
+
+export default Day;
+
+const StyledDay = styled.div`
+  display: flex;
+`;
+
+const DayString = styled.h2`
+  color: #f9a11b;
```
Comment (plusk) [inline src/components/timeline/Day.tsx]: Should use our actual colors now that they're merged into master

File: src/components/timeline/Event.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+  const toggleExpanded = () => {
+    setExpanded(!expanded);
+  };
+
+  return (
+    <StyledEvent>
+      <EventHeader onClick={toggleExpanded}>
+        <EventTime>{startTime}</EventTime>
+        <EventIndicator className="indicator" />
```
Comment (plusk) [inline src/components/timeline/Event.tsx]: Using a class could be problematic on principle, as it won't be hashed or anything. That way, it technically speaking could be targetted by other styling. Even if that is highly unlikely, there are other (safer) ways we can do this styling

File: src/components/timeline/Event.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    </StyledEvent>
+  );
+};
+
+export default Event;
+
+const StyledEvent = styled.div`
+  &:hover {
+    .indicator {
+      background-color: orange;
```
Comment (plusk) [inline src/components/timeline/Event.tsx]: Use the color palette

File: src/components/timeline/Event.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    }
+  }
+`;
+
+const EventTitle = styled.h3`
+  font-weight: bold;
+  border-left: 1px solid black;
```
Comment (plusk) [inline src/components/timeline/Event.tsx]: Use the color palette

File: src/components/timeline/Day.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  font-weight: 400;
```
Comment (plusk) [inline src/components/timeline/Day.tsx]: Very nitpicky, but I'd prefer using `font-weight: normal`. For example, the keyword "bold" means 700, but uses [fallback weights](https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight#Fallback_weights) if it isn't found. The specific font weight numbers we import could vary, so using keywords ensures you'll get "something bold-ish" even if specifically 700 isn't available.

File: src/components/timeline/Event.tsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    margin: 0px;
+  }
+`;
+
+const EventTime = styled.h3`
+  padding: 16px 30px 16px 0;
+`;
+
+const EventEndTime = styled.p`
+  font-size: 14px;
```
Comment (plusk) [inline src/components/timeline/Event.tsx]: I'd recommend using breakpoints for this component, so you can specify font-size for the given "resolution range", if you will. By default our body size should maybe stick to the default of 16px, but that's just me guessing it's a default for a good reason as far as legibility goes. The underline makes this even harder to read the tinier it is also

File: src/components/timeline/Event.tsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+  text-decoration: underline;
+  padding-top: 16px;
+`;
+
+const EventBody = styled.div<{ hide?: boolean }>`
+  border-left: 1px solid black;
```
Comment (plusk) [inline src/components/timeline/Event.tsx]: :art:

File: src/components/timeline/Event.tsx [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+  ${({ hide }) =>
+    hide &&
+    `
+    display: none;
+  `}
+`;
+
+const EventInnerContent = styled.div`
+  border-bottom: 1px solid transparent;
```
Comment (plusk) [inline src/components/timeline/Event.tsx]: While I was the one who talked about this thing, are we sure there isn't a better way to solve the issue of collapsing margins? Structuring the HTML differently might help

File: src/components/timeline/Event.tsx [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+  & > * + * {
+    margin: 16px 0;
+  }
+`;
+
+const EventIndicator = styled.div`
+  position: absolute;
+  left: 83px;
```
Comment (plusk) [inline src/components/timeline/Event.tsx]: I think having the indicator as a div might prove to be problematic. First of all, I'm not a huge fan of having HTML elements that are only there for styling. This is a great fit for a pseudo-element, for example.
Being pixel-specific here leads to the dot being positioned incorrectly if the size or weight of the fonts nearby are changed. A more general solution could be using `&::before` on the EventTitle, as that is where the border we want the dot to sit on is defined as well. Positioning it based on the EventTitle shouldn't be too hard either, something like `left: 0` and maybe a `transform: translateX(-50%)` could get you where you want to be.

File: src/components/timeline/Event.tsx [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+  top: 20px;
+  background-color: black;
```
Comment (plusk) [inline src/components/timeline/Event.tsx]: :art:

File: src/components/timeline/Timeline.stories.mdx [hunk 11]
```diff
@@ -0,0 +1,73 @@
+import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
+import Timeline from './Timeline';
+import Day from './Day';
+import Event from './Event';
+
+<Meta title="Others|Timeline" component={Timeline} />
+
+const event = Event
```
Comment (plusk) [inline src/components/timeline/Timeline.stories.mdx]: This line doesn't do anything

File: src/components/timeline/Timeline.stories.mdx [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+            Vi feirer alle nye studenter som er immatrikulert med tradisjonsrikt ball! Børst støvet av finstasen og gjør
+            deg klar for immball!
+          </p>
+        </Event>
+      </Day>
+    </Timeline>
+  </Story>
+</Preview>
+
+<Props of={Timeline} />
```
Comment (plusk) [inline src/components/timeline/Timeline.stories.mdx]: Timeline doesn't have any props, but showing props for Day and Event could be useful

File: src/components/timeline/Day.tsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+import styled from 'styled-components';
+import { colors } from 'common/colors';
+import { media } from 'common/layout';
+
+interface DayProps {
+  children: React.ReactNode;
+  day: string;
+}
+
+const Day = ({ children, day }: DayProps): JSX.Element => {
```
Comment (oleast) [inline src/components/timeline/Day.tsx]: We took a little cleanup of the eslint rules and decided to remove the requirement of defining `JSX.Element`.

File: src/components/timeline/Timeline.tsx [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+
+interface TimelineProps {
+  children: React.ReactNode;
+}
+
+const Timeline = ({ children }: TimelineProps) => {
+  return <>{children}</>;
+};
+
+export default Timeline;
```
Comment (oleast) [inline src/components/timeline/Timeline.tsx]: What is the purpose of this component?

File: src/components/timeline/Event.tsx [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+import { colors } from 'common/colors';
+
+interface EventProps {
+  title: string;
+  startTime: string;
+  endTime: string;
+  children: React.ReactNode;
+}
+
+const Event = ({ title, startTime, endTime, children }: EventProps): JSX.Element => {
```
Comment (oleast) [inline src/components/timeline/Event.tsx]: Same thing with `JSX.Element`

File: src/components/timeline/Timeline.tsx (hunk 14 above)
Comment (aslakhol) [reply src/components/timeline/Timeline.tsx]: It makes more semantic sense to for a user of the component to instantiate a <Timeline> component, so I decided to keep this for the usability of the component in later projects.

**Pull request merged**